$NetBSD: patch-ae,v 1.4 2018/02/05 18:37:21 jperkin Exp $

let <string.h> declare strcasestr() correctly
Fix fpos_t cast on NetBSD (_pos vs. __pos)

--- include/mpeg4ip.h.orig	2006-08-07 18:27:00.000000000 +0000
+++ include/mpeg4ip.h
@@ -120,14 +120,6 @@
 #endif
 #include <sys/param.h>
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-char *strcasestr(const char *haystack, const char *needle);
-#ifdef __cplusplus
-}
-#endif
-
 #define OPEN_RDWR O_RDWR
 #define OPEN_CREAT O_CREAT 
 #define OPEN_RDONLY O_RDONLY
@@ -156,6 +148,9 @@ char *strcasestr(const char *haystack, c
 #ifdef HAVE_FPOS_T___POS
 #define FPOS_TO_VAR(fpos, typed, var) (var) = (typed)((fpos).__pos)
 #define VAR_TO_FPOS(fpos, var) (fpos).__pos = (var)
+#elif defined(__NetBSD__)
+#define FPOS_TO_VAR(fpos, typed, var) (var) = (typed)((fpos)._pos)
+#define VAR_TO_FPOS(fpos, var) (fpos)._pos = (var)
 #else
 #define FPOS_TO_VAR(fpos, typed, var) (var) = (typed)(fpos)
 #define VAR_TO_FPOS(fpos, var) (fpos) = (var)
@@ -194,7 +189,7 @@ typedef uint16_t in_port_t;
 typedef unsigned int socklen_t;
 #endif
 
-#ifdef sun
+#ifdef __sun
 #include <limits.h>
 #define u_int8_t uint8_t
 #define u_int16_t uint16_t
@@ -233,7 +228,8 @@ char *strsep(char **strp, const char *de
 
 #define ADV_SPACE(a) {while (isspace(*(a)) && (*(a) != '\0'))(a)++;}
 
-#ifndef HAVE_GTK
+#ifndef __G_LIB_H__
+/* #ifndef HAVE_GTK */
 typedef char gchar;
 typedef unsigned char guchar;
 
